internal/git/log.go

```diff
@@ -0,0 +1,93 @@
+package git
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	closeCommitPattern = regexp.MustCompile(`(?i)\b(?:close|closes|closed|fix|fixes|fixed|resolve|resolves|resolved)\W+#(\d+)\b`)
+)
+
+type LogOpts struct {
+	// RevisionRange is the range of the commits specified by the format described in
+	// git-log(1).
+	RevisionRange string
+}
+
+// Log returns a list of commits specified by the range.
+func (r *Repo) Log(opts LogOpts) ([]*CommitInfo, error) {
+	res, err := r.Run(&RunOpts{
+		Args:      []string{"log", "--format=%H%x00%h%x00%s%x00%b%x00", opts.RevisionRange},
```

Comment:
Can you add a trailing `--` here to avoid the possibility of an ambiguous revision/path error? According to manpages, `git log` can take paths as well.

Reply:
Done.